src/main/java/org/smartregister/converters/StockConverter.java

```diff
@@ -54,46 +55,56 @@ public static Bundle convertStockToBundleResource(StockAndProductDetails stockAn
 		deviceBuilder.owner(owner).location(location).serialNumber(serialNumber)
 				.status(fhirDeviceStatus);
 
-		SupplyDelivery.Builder supplyDeliveryBuilder = SupplyDelivery.builder();
-		Decimal decimal = Decimal.builder().value(java.lang.String.valueOf(stockAndProductDetails.getStock().getValue()))
-				.build();
-		SimpleQuantity simpleQuantity = SimpleQuantity.builder().value(decimal).build();
+		List<Bundle.Entry> entryList = new ArrayList<>();
+		java.lang.String productId =
+				stockAndProductDetails != null && stockAndProductDetails.getProductCatalogue() != null &&
+						stockAndProductDetails.getProductCatalogue().getUniqueId() != null ?
+						java.lang.String.valueOf(stockAndProductDetails.getProductCatalogue().getUniqueId()) : "";
 
-		if (stockAndProductDetails != null && stockAndProductDetails.getProductCatalogue() != null &&
-				stockAndProductDetails.getProductCatalogue().getProductName() != null) {
-			deviceItemCode = Code.builder().value(stockAndProductDetails.getProductCatalogue().getProductName())
-					.build();
-			deviceItemCoding = Coding.builder().code(deviceItemCode)
-					.display(String.of(stockAndProductDetails.getProductCatalogue().getProductName())).build();
-			deviceItemElement = CodeableConcept.builder().coding(deviceItemCoding).build();
+		java.lang.String deviceUriString = "https://fhir.smartregister.org/device/" + productId;
```

Comment:
I think that this URL should not be hardcoded. Does it have any function currently

Reply:
Nope